README.md

```diff
@@ -383,6 +383,41 @@ r.Walk(func(route *mux.Route, router *mux.Router, ancestors []*mux.Route) error
 })
 ```
 
+### Middleware
+
+
+Since **vX.Y.Z**, mux supports the addition of middlewares to a router, which are executed if a
+match is found (including subrouters). Middlewares are defined using the de facto standard function:
+
+```go
+func(http.Handler) http.Handler
+```
+
+Structs that implement the `Middleware` interface can also be used:
+
+```go
+type Middleware interface {
```

Comment:
I'm not entirely convinced this interface is useful. Can you expand more on how this is a better API vs. only supporting `func(http.Handler) http.Handler` - ?